portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
```

Comment:
curious, I assume here should be also a ```doPost`` to handle users login requests?
Or you want to implement login in the following PRs?

Reply:
I think a login link will be returned by `response.getWriter().println(USERSERVICE.createLoginURL("/index.html"));`
And when user click on the link, login will be handled by the authentication library. There will be some interface like this
<img width="1387" alt="Screen Shot 2020-06-28 at 5 42 56 PM" src="https://user-images.githubusercontent.com/37273127/85944160-4cd4b580-b967-11ea-8556-de6ffaeaaf16.png">